### PR TITLE
MM-17458 Integration test for system message on header change

### PR DESCRIPTION
**Pull request by ashishmahi**
Summary

Add integration test for text box height not exceeding 50%
Ticket Link

Fixes: mattermost/mattermost-server#11754
Jira : https://mattermost.atlassian.net/browse/MM-17458

### CODE REVIEW

File: e2e/cypress/integration/channel/header_spec.js [hunk 1]
```diff
@@ -81,6 +81,40 @@ describe('Header', () => {
...
+        before(() => {
+            // # cheange channel header
+            cy.updateChannelHeader('updated header');
+        });
+
+        beforeEach(() => {
+            // # hover on last message i.e system message
+            cy.get('.post--system').last().trigger('mouseover');
+        });
+
```
Comment (srkgupta) [inline e2e/cypress/integration/channel/header_spec.js]: We generally have only one describe, before and beforeEach methods for each test file.

Comment (srkgupta) [reply e2e/cypress/integration/channel/header_spec.js]: Also please mention the test key and title as mentioned in this ticket https://mattermost.atlassian.net/browse/MM-17458
*Test Key:* M17458
*Title:* System message limited options

File: e2e/cypress/integration/channel/header_spec.js [hunk 2]
```diff
@@ -81,6 +81,40 @@ describe('Header', () => {
...
+                should('not.be.visible');
+        });
+        it('should have delete button on system message', () => {
+            // # verify that delete option is available on system message
+            // cy.get('.post--system').last().trigger('mouseover');
+            cy.get('button').
+                contains('Delete').
+                should('be.visible');
+        });
+    });
```
Comment (srkgupta) [inline e2e/cypress/integration/channel/header_spec.js]: Individual it statements are not required. The comments added are good enough

File: e2e/cypress/integration/channel/header_spec.js [hunk 3]
```diff
@@ -81,6 +81,40 @@ describe('Header', () => {
         // # Verify that the Search term input box is still cleared and search term does not reappear when RHS opens
         cy.get('#searchBox').should('be.visible').and('be.empty');
     });
+
+    describe('Header update', () => {
+        before(() => {
+            // # cheange channel header
```
Comment (srkgupta) [inline e2e/cypress/integration/channel/header_spec.js]: Suggested change:
```// # change channel header```

File: e2e/cypress/integration/channel/header_spec.js [hunk 4]
```diff
@@ -81,6 +81,40 @@ describe('Header', () => {
...
+        it('should not have pin to channel message ', () => {
+            // # verify that pin_to_channel button is  not visible on system message
+            cy.clickPostDotMenu().get('button').
+                contains('Pin to channel').
+                should('not.be.visible');
+        });
+        it('should have delete button on system message', () => {
+            // # verify that delete option is available on system message
+            // cy.get('.post--system').last().trigger('mouseover');
+            cy.get('button').
```
Comment (srkgupta) [inline e2e/cypress/integration/channel/header_spec.js]: Please use an exact unique identifier. Please follow the recommended practice as mentioned here to provide context to the selectors. 
https://developers.mattermost.com/contribute/webapp/end-to-end-tests/#writing-specs

Comment (srkgupta) [reply e2e/cypress/integration/channel/header_spec.js]: Similarly applicable for other selectors as well like .flag-icon__container (or) .reaction__container

File: e2e/cypress/integration/channel/header_spec.js (hunk 1 above)
Comment (ashishmahi) [reply e2e/cypress/integration/channel/header_spec.js]: as discussed changes are done and pushed

**Comment by saturninoabril** [comment]
@ashishmahi Looks like you've pushed unrelated commits.  Please remove those and only include those you've created.

**Comment by saturninoabril** [comment]
@ashishmahi Let me know if you need help on this.

**Comment by saturninoabril** [comment]
@ashishmahi I'm closing this due to inactivity.  Feel free to submit a separate PR.  Thanks for looking into it.

**Pull request closed without merging**